backend/agent_logic/tools/sandbox_executor.py

```
@@ -1,79 +1,257 @@
+        except docker.errors.ImageNotFound as e:
+            logger.error(f"? Docker image not found: {str(e)}")
+            return {
+                "success": False,
+                "output": "",
+                "error": f"Docker image not found: {str(e)}",
+                "exit_code": -1
+            }
+        
         except Exception as e:
-            logger.error(f"Sandbox execution error: {str(e)}")
+            logger.error(f"? Sandbox execution error: {str(e)}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)